scripts/setup/setup-certbot

```diff
@@ -93,8 +97,9 @@ chmod a+x "$CERTBOT_PATH"
 # to agree to the Let's Encrypt Subscriber Agreement (aka ToS).
 # Passing --force-interactive suppresses a warning, but also brings up
 # an annoying prompt we stifle with --no-eff-email.
+# shellcheck disable=SC2086
 "$CERTBOT_PATH" certonly "${method_args[@]}" \
-                -d "$DOMAIN" -m "$EMAIL" \
+                ${HOSTNAMES[*]} -m "$EMAIL" \
```

Comment:
And this should be `"${HOSTNAMES[@]}"` (including double quotes, to avoid splitting the array members on whitespace). ShellCheck is correct here; don’t disable it. Perhaps valid hostnames won’t have whitespace, but we’d still prefer the failure mode for invalid hostnames to make some amount of sense—and in general, after I just put lots of effort into fixing the shell scripts that played fast and loose with whitespace, I don’t want more of them to start creeping in for no reason.

Reply:
Furthermore, valid hostnames _may_ be wildcards starting with `*`, so we do need to avoid glob expansion.

Reply:
Using double quotes here was ending up in the output looking like:

`/certbot '-d domain.name -d another.name' --email something.....`

Emphasis on the additional single quotes being inserted. I'm not sure how to avoid that.

Reply:
`"${HOSTNAMES[@]}"` really is the syntax you're looking for. Make sure you have the double quotes and the `@` instead of `*`.